### PR TITLE
fix(designer-ui): Action icons with no connector display name have bad ARIA text

### PR DESCRIPTION
**Pull request by ek68794998**
- **Please check if the PR fulfills these requirements**

* [x] The commit message follows our guidelines
* [x] Tests for the changes have been added (for bug fixes/features)
* [ ] Docs have been added / updated (for bug fixes / features)

- **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

Bug fix

- **What is the current behavior?** (You can also link to an open issue here)

Connector icon ARIA text works fine:

image

But if no connector is defined in the manifest, the fallback is not good:

image

- **What is the new behavior (if this is a feature change)?**

Both work correctly, using the operation name (i.e., operation display name, **not** the user-inputted name) if there is no connector name:

image

image

- **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)

No

- **Please Include Screenshots or Videos of the intended change**:

See above

### CODE REVIEW

File: libs/designer-ui/src/lib/card/index.tsx [hunk 1]
```diff
@@ -105,30 +107,40 @@ export const Card: React.FC<CardProps> = ({
...
+          defaultMessage: '{connectorName} connector icon',
+          description: 'Alt text icon on action/trigger card when there is a connector name',
+        },
+        {
+          connectorName,
+        }
+      )
+    : intl.formatMessage(
+        {
+          defaultMessage: '{operationName} icon',
```
Comment (ek68794998) [inline libs/designer-ui/src/lib/card/index.tsx:122]: Wasn't sure if this should be `___ icon` or `___ operation icon`?

**Comment by hartra344** [comment]
I actually wonder if an argument could be made that these images are decorative? And maybe shouldn't have alt text at all? Is this the only way we indicate the connector type? Maybe a better solution would be to add screen reader hints to tell the connector type and the icon doesn't matter.

W3C guidance on this: https://www.w3.org/WAI/tutorials/images/decorative/

**Comment by ek68794998** [comment]
> I actually wonder if an argument could be made that these images are decorative? And maybe shouldn't have alt text at all? Is this the only way we indicate the connector type? Maybe a better solution would be to add screen reader hints to tell the connector type and the icon doesn't matter.
> 
> W3C guidance on this: https://www.w3.org/WAI/tutorials/images/decorative/

We used to do so, but bug #3131 (resolved in PR #3145) was an accessibility sev-2 on us that indicated they should be announced rather than seen as decorative. Thoughts?

**Comment by hartra344** [comment]
I think that's a valid bug if it's the only way we indicate connector type. However I think a better fix is to make connector type better accessable(even for non screen reader tech hoestly, who memorizes connector by icon) than to make decorative icons screen redundantly read out in a way that doesn't matter.

If I'm seeing impaired, or fully blind, what does "Outlook connector icon" mean to me? The only valuable information I'm getting out of that is "Outlook Connector". I likely don't know what the icon looks like and why would I care that it's there? It's extra information that does nothing.

**Review by Eric-B-Wu — APPROVED** [review]

**Comment by ek68794998** [comment]
> I think that's a valid bug if it's the only way we indicate connector type. However I think a better fix is to make connector type better accessable(even for non screen reader tech hoestly, who memorizes connector by icon) than to make decorative icons screen redundantly read out in a way that doesn't matter.
> 
> If I'm seeing impaired, or fully blind, what does "Outlook connector icon" mean to me? The only valuable information I'm getting out of that is "Outlook Connector". I likely don't know what the icon looks like and why would I care that it's there? It's extra information that does nothing.

This is a valid point. Is it perhaps better to read the card out as "SharePoint, Create item card, button" or similar, rather than trying to read the icon? I'm not sure if the accessibility team's concern here is that the icon needs to be readable, or whether the card button doesn't have enough ARIA content for screen readers.

**Comment by hartra344** [comment]
> Member

I like this idea, the only change would be that I would go "Action Name, Connector" since action name is the primary information (button announcement should come from role).

**Comment by ek68794998** [comment]
Sounds good, I will look at that when I have cycles. Dropping this back to draft for now.

**Comment by hartra344** [comment]
And if they a11y team wants to push adding a description to the icon, I'm more than happy to go to bat on that. It goes directly against W3C guidance, likely coming from automated tooling without any thought behind if it's right or not.

File: libs/designer/src/lib/core/state/selectors/actionMetadataSelector.ts [hunk 2]
```diff
@@ -107,40 +109,53 @@ const useNodeAttribute = (operationInfo: NodeOperation, propertyInManifest: stri
...
   const useManifest = operationManifestService.isSupported(operationInfo?.type ?? '', operationInfo?.kind ?? '');
 
-  const { data: connectorData } = useConnectorAndSwagger(operationInfo?.connectorId, !useManifest);
-  const { result, isLoading } = useNodeAttribute(operationInfo, ['connector', 'properties', 'externalDocs'], ['externalDocs']);
+  const result = useNodeAttributeOrSwagger(operationInfo, ['summary'], ['summary'], 'summary', { useManifest });
+  if (result.result === undefined && operationInfo?.operationId) {
+    result.result = titleCase(operationInfo.operationId);
+  }
+
+  console.log(operationInfo, result);
```
Comment (ek68794998) [inline libs/designer/src/lib/core/state/selectors/actionMetadataSelector.ts:141]: Need to remove this.

**Review by Eric-B-Wu — APPROVED** [review]

**Comment by hartra344** [comment]
> @hartra344 Found some time between meetings to update the PR :) Let me know if this meets your expectations.
> 
> Edit: My only confusion was which of these `<div>`s should have `role=button`:
> 
> image

Probably just the one that has focus. I think only the card itself gets focus, inside the card does not but would be good to check.

**Comment by ek68794998** [comment]
> > @hartra344 Found some time between meetings to update the PR :) Let me know if this meets your expectations.
> > Edit: My only confusion was which of these `<div>`s should have `role=button`:
> > image
> 
> Probably just the one that has focus. I think only the card itself gets focus, inside the card does not but would be good to check.

Sounds good. I just validated that Narrator properly reads out when tabbing between cards using the latest build.

image

**Pull request merged**